UI/src/view/point_asset/trafficSignForm.js

```diff
@@ -413,19 +431,57 @@
         '      </select></div>';
     };
 
-    var singleChoiceTrafficSignTypeHandler = function (property, collection) {
+
+    function isPedestrianOrCyclingRoadLink(asset) {
+      if (asset === undefined){
+        return false;
+      }
+
+      var roadLink = me.roadCollection.getRoadLinkByLinkId(asset.linkId);
+      var roadLinkData;
+
+      if (roadLink){
+        roadLinkData = roadLink.getData();
+        if (roadLinkData !== undefined && roadLinkData.linkType === 8) {
+          return true;
+        }
+      }
+      return false;
+    }
```

Comment:
This function could be simplified like this
```javascript
if(asset){
      var roadLink = me.roadCollection.getRoadLinkByLinkId(asset.linkId);

      if (roadLink){
        var roadLinkData = roadLink.getData();
        return !_.isUndefined(roadLinkData) && roadLinkData.linkType === 8;
      }
}
return false;
```

Reply:
Applied both suggestions.
roadLinkData.linkType === 8 will be moved to roadCollection.js